### DISCUSSION TITLE
How to post-process a saved video file?

### DISCUSSION THREAD

**Discussion by compilebunny** [bluenviron/mediamtx#4089] [Questions]
### Question

I am using mediamtx to save RTSP streams from a camera to a video file. MP4 files are generated using 

```
  # Record streams to disk.
  record: yes
  playback: no
  recordPath: /files/video%whatever
  recordFormat: fmp4  
  recordSegmentDuration: defined_time_period
```

I want to post-process each resulting mp4 file as it is completed. I tried using 

runOnRecordSegmentComplete: /bin/post-processing-program

but /bin/post-processing-program never gets run. 

How can I post-process each video file that is recorded to disk?

**Comment by aler9** [comment]
The solution is to use `runOnRecordSegmentComplete`. If the command does not get run, open a bug report and provide precise instructions on how to replicate the issue.
